apps/workflow-server/gcc-workflow-server/src/main/java/com/coremedia/labs/translation/gcc/workflow/GlobalLinkAction.java

```diff
@@ -316,8 +344,10 @@ public final ActionResult storeResult(Task task, Object result) {
     @SuppressWarnings("unchecked" /* per interface contract: result is the return value of #doExecute */)
     Result<R> r = (Result<R>) result;
 
+    LOG.info("action result retries {}, delay {}", r.remainingAutomaticRetries, r.retryDelaySeconds);
     Process process = task.getContainingProcess();
     process.set(remainingAutomaticRetriesVariable, r.remainingAutomaticRetries);
+    process.set(retryDelayTimerVariable, new RelativeTimeLimit(r.retryDelaySeconds));
```

Comment:
Saw that and found it disturbing since the same holds for _remainingAutomaticRetriesVariable_ and _issues_ - neither of those is checked here. If I add a check, I should probably do it for all three of them, right?

And what would I do, if any of these vars is _null_? Throwing an exception is what would happen right now, too. This already informs the developer about the missing (but logically required) variable. And this information would not show up, if I guard the setter with a _null_ check. 

Or am I getting this wrong?

Reply:
setRetryDelayTimerVariable and its corresponding attribute in translation-global-link.xml were introduced with a later version of this integration than the other setters. For such a version update, the workflow definition needs to be re-uploaded. But there may still be running workflows that use the previous definition, and those workflows would use the new java code but retryDelayTimerVariable would still be null, because the GlobalLinkAction is deserialized from data of the WFS database. The null-check is required to prevent issues when upgrading. If the variable name is not set, then the variable doesn't need to be set, because it also doesn't exist in the old definition. But the java code must still work with the old data and definition.

I think we should not drop such existing checks, to still allow upgrades from very old versions.

Null-checking the other attributes could theoretically also make sense, but it's not that important. (They could be checked in #extractParameters to let workflows fail early.)

Reply:
Got it now - will change.